internal/build/helper.go

```diff
@@ -7,20 +7,20 @@ import (
 	ootov1alpha1 "github.com/qbarrand/oot-operator/api/v1alpha1"
 )
 
-//go:generate mockgen -source=modulehelper.go -package=build -destination=mock_modulehelper.go
+//go:generate mockgen -source=helper.go -package=build -destination=mock_helper.go
 
-type ModuleHelper interface {
+type Helper interface {
 	ApplyBuildArgOverrides(args []v1alpha1.BuildArg, overrides ...v1alpha1.BuildArg) []v1alpha1.BuildArg
 	GetReleventBuild(mod ootov1alpha1.Module, km ootov1alpha1.KernelMapping) *ootov1alpha1.Build
 }
 
-type moduleHelper struct{}
+type helper struct{}
 
-func NewModuleHelper() ModuleHelper {
-	return moduleHelper{}
+func NewHelper() Helper {
```

Comment:
I think every `New*` function in the code should return a struct and not an interface.
The "thumb rule" in golang says "accept interfaces, return structs"

Here are a few reasons for that:
* By returning an interface, we restrict the user of the package to access public fields in the struct
* The "user" should be able to decide if the returned object is stored as an interface or as a struct (more flexibility). In case the user only want the interface as return value, we should use `var helper Helper = NewHelper()` but not restrict it at the library level.
* What will be returned in case an object implement multiple interfaces (even if we don't have such in the code ATM)? This is not working when returning interfaces.

Reply:
I agree with the points that you mentioned here, and will change the code, but there are 3 counter-arguments:
1) by returning interface you are restricting the usage of your package. User can only use the package through interface that you defined. That, in turn, makes it safer for you to change the implementation code without worrying that you can break user code
2) by returning interface m you are actually checking that your stuct has implemented all the functions defined by the interface during compilation time. Otherwise it will be checked only when user code is compiled, and only in case he uses it as an interface
3) it is not relevant for our code, but in case we return pointer instead of interface, the inner structure must be made public, otherwise the variable cannot be defined explicitly

Reply:
Thanks. Just for more clarification:

> by returning interface you are restricting the usage of your package. User can only use the package through interface that you defined. That, in turn, makes it safer for you to change the implementation code without worrying that you can break user code

Isn't that what private methods/fields are for? By setting a public field, this is treated as API and shouldn't be changed (same as we "don't" change interface methods).

> by returning interface m you are actually checking that your stuct has implemented all the functions defined by the interface during compilation time. Otherwise it will be checked only when user code is compiled, and only in case he uses it as an interface

+1 on that. Although I think that UT should cover this one but it is indeed better to have the compiler check it.

> it is not relevant for our code, but in case we return pointer instead of interface, the inner structure must be made public, otherwise the variable cannot be defined explicitly

Actually this one is a very good point!